### PR TITLE
Javascript automated testing and a11y doc

### PR DESCRIPTION
**Pull request by jacobherrington**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [x] Feature?
- [ ] Bug Fix
- [ ] Documentation Update

## Description

I've spent a non-trivial amount of time learning about accessibility over the last few weeks. It's not something I've done much work around in past (neither is JavaScript testing!), but I'm planning to expand these tests and try to get some basic protection in place for regressions in accessibility as I look for opportunities to improve the experience we currently have.

## Added to documentation?

- [x] docs.dev.to
- [ ] readme
- [ ] no documentation needed

## [optional] What gif best describes this PR or how it makes you feel?

TEST

### CODE REVIEW

File: docs/tests/accessibility-tests.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+identify some of the potential accessibility concerns on DEV.
+
+Many a11y issues are not obvious to everyone who contributes to the DEV project,
+therefore leaning on tools to help us identify these issues is a good practice.
+
+It's a good idea to use browser plugins while you're developing to keep an eye
+out for these issues, as well as including automated tests to catch regressions
+in future changes.
+
+The [aXe devtools](https://www.deque.com/axe/extension) is a great place to get
```
Comment (benhalpern) [inline docs/tests/accessibility-tests.md]: I'm getting a 404 at that link.

Comment (jacobherrington) [reply docs/tests/accessibility-tests.md]: u right

Good catch, just fixed it! :sweat_smile:

**Comment by nickytonline** [comment]
Nice work Jacob!

+1 for proposing LightHouse CI @rhymes. I've been reading up on it. They have a GitHub action for it, so that would definitely be something interesting to add to CI. I think that will help on many fronts, not just a11y.

Also, there are already some [a11y eslint rules](https://github.com/thepracticaldev/dev.to/blob/master/app/javascript/.eslintrc.js#L3) set up, but adding axe is definitely a great idea.

They have an extension for [Chrome](https://chrome.google.com/webstore/detail/axe-web-accessibility-tes/lhdoppojpmngadmnindnejefpokejbdd?hl=en-US) and [FireFox](https://addons.mozilla.org/en-CA/firefox/addon/axe-devtools/) as well.

**Comment by jacobherrington** [comment]
@rhymes I hope we can have a larger discussion as a team before moving forward, but this gives a baseline. That's why this PR only tests a single component (and honestly only a piece of it). Ideally, we can expand this gradually to fix existing a11y issues and introduce a protection against regressions that will throw an error during CI. I think I'm on the same page.

@nickytonline I spent quite a bit of time looking into CI tools, and honestly got in a bit over my head before doing this PR. I think those tools *might* be overcomplicating the issue. Of course, I'm much less experienced in this arena than you. I had some problems reconciling a few of the CI tools I found with Rails. It doesn't seem like many tools exist to check for a11y issues with Rails apps. However, it's possible that this is the wrong route and going with Lighthouse CI would be easy. :man_shrugging:

**Pull request merged**